plugins/flytekit-kf-mpi/flytekitplugins/kfmpi/task.py

```diff
@@ -133,5 +131,70 @@ def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
         return MessageToDict(job.to_flyte_idl())
 
 
+@dataclass
+class HorovodJob(object):
+    slots: int
+    num_launcher_replicas: int = 1
+    num_workers: int = 1
+
+class HorovodFunctionTask(PythonFunctionTask[HorovodJob]):
```

Comment:
Could we inherit `MPIFunctionTask` instead?

Reply:
I'm not sure how to do so.
Using `HorovodFunctionTask(PythonFunctionTask[HorovodJob]):` can enable @task to [find_pythontask_plugin](https://github.com/flyteorg/flytekit/blob/master/flytekit/core/task.py#L203) correctly, and use horovod command.

If inherit from `mpifunctiontask`, how @task distinguish `mpifunctiontask` and `horovodfunctiontask`

Reply:
like:
```python
class HorovodFunctionTask(MPIFunctionTask)
     ...
TaskPlugins.register_pythontask_plugin(HorovodJob, HorovodFunctionTask)
```

Reply:
Oh thanks! Just realize @task uses `TaskPlugins.register_pythontask_plugin(HorovodJob` to find the right task instead of `HorovodFunctionTask(PythonFunctionTask[HorovodJob]):`

Reply:
But anyways we still need another dataclass for horovod right?